src/compile/axis.ts

```diff
@@ -177,22 +178,32 @@ export namespace properties {
       }, labelsSpec);
     }
 
+    let textTemplate = '{{ datum.data }}';
+
     if (contains([NOMINAL, ORDINAL], fieldDef.type) && axis.labelMaxLength) {
       // TODO replace this with Vega's labelMaxLength once it is introduced
-      labelsSpec = extend({
-        text: {
-          template: '{{ datum.data | truncate:' + axis.labelMaxLength + '}}'
-        }
-      }, labelsSpec || {});
+      textTemplate = '{{ datum.data | truncate:' + axis.labelMaxLength + '}}';
```

Comment:
I need to figure out how vega uses these text templates. Because the text right now looses the date/time formatting.

Also on a side note shouldn't there be a unit test that checks if the units are being formatted for date/time